lib/ArduinoNative/Arduino.h

```diff
@@ -156,8 +161,39 @@ class Serial_
         printf("%d\n", value);
     }
 
-private:
+    void write(const uint8_t* buffer, size_t length)
+    {
+        m_socketServer.sendMessage(buffer, length);
+    }
+
+    int available()
+    {
+        return m_socketServer.available();
+    }
 
+    size_t readBytes(uint8_t* buffer, size_t length)
+    {
+        size_t count = 0;
+
+        for (count = 0; count < length; count++)
+        {
+            uint8_t byte;
+            if (!m_socketServer.getByte(byte))
+            {
+                break;
+            }
+            else
+            {
+                buffer[count] = byte;
+            }
+        }
+
+        return count;
+    }
+
+private:
+    /** Instance of the Socket Server */
+    SocketServer m_socketServer;

```

Comment:
The class is mixed. Hmm .... would be better to have it dedicated and decided somewhere else when to use the socket server. Keep in mind that really every application will get this.

Reply:
I do not believe there is an usecase where the application running on the simulation will not use the YAP Server, and therefore it requires the SocketServer to send the data. But it could still be separated for better maintainability. I would propose separating the Serial_ class into its own header and source files, and the implementation in the source file can decide if/when the SocketServer will be used

Reply:
I think about further applications in future, but it makes sense to separate it and support all Serial features, e.g. all print's should be send via socket too. In the test enviroment we could have again a separate TestSerial. Good idea.